### PR TITLE
`Development`: Use artemisApp as consistent prefix for translation keys

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).

### Motivation and Context
@Santia-go noticed that modeling exercises don't use the artemisApp prefix for its translation key. This PR fixes this inconsistency for multiple cases.

### Description
There are still a few files left that don't start with artemisApp. These are files like home, error, alert, etc. which are not direct features of Artemis. All other files that refer to Artemis features should now use the artemisApp prefix.

### Steps for Testing
1) Pease review the codebase to check for places where I could have missed some translations. 

2) Test the following components and check if all strings are correctly displayed in both German and English. 

- User Management
- Organisation Management
- Modeling Exercises (the whole lifecycle, participating, assessment, complaints)
- Quiz-Statistics

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### CODE REVIEW

**Review by Dominik-Weinzierl — COMMENTED** [review]
The changes looks good to me :thinking: but instead of adding all the prefixes, we could also just remove them :thinking: are they really required?

**Review by rhubinger — APPROVED** [review]
Tested on TS2. Everyting works and translations look good.

**Review by SimonPaul02 — DISMISSED** [review]
Tested on TS2. Everything works as expected

**Review by yuling-sun — APPROVED** [review]
Tested on TS2. Strings look fine to me.

**Review by Santia-go — APPROVED** [review]
Tested on TS1
I have tested the functionality of course, exam, exercises, and shared files. 
They are working correctly. The corresponding tests look good.

**Review by Santia-go — DISMISSED** [review]
ReTested on TS1
User and Organisation Management works fine
All around the Modeling Exercises works well also
No errors in Quiz-Statistics

**Review by Marvin-Vitcu — CHANGES_REQUESTED** [review]
When using the filter in User Management. Some translations are not found:

englisch version:
image

german version:
image

**Review by Marvin-Vitcu — APPROVED** [review]
Tested on TS1. The User Management Filter now displays the correct translation.

**Review by max-bergmann — APPROVED** [review]
Tested on TS2. Works as expected

**Review by SimonPaul02 — DISMISSED** [review]
Retested on TS1👍

**Review by yuling-sun — DISMISSED** [review]
Re-tested on TS1. Translations look good.

**Review by nairaarnez — CHANGES_REQUESTED** [review]
Tested in TS1, there are some small details:
image
Ich würde es zu "erweiterte Suche (an?)zeigen" umschreiben
image
"Code Hinweise generieren", "Unterschied zwischen Template und Lösung zeigen" und " strukturelle Solution Entries erstellen"
image
"Hinweise Verwalten"
image
to increase consistency in the writing. All verbs in the end (I would suggest this) or at the beginning (imperative)

**Comment by Strohgelaender** [comment]
@nairaarnez Thanks for the review. This PR is only about refactoring the translation keys. I don't want to change the values as well since this PR is already quite big and should not get even bigger.

**Review by JohannesStoehr — APPROVED** [review]
Code looks good after the merge 👍

**Review by nairaarnez — APPROVED** [review]
Tested in ts1, looks good, approved!

**Review by ge65cer — COMMENTED** [review]
We should probably merge https://github.com/ls1intum/Artemis/pull/5296/ before this, so error messages also have a consistent `artemisApp` prefix?

**Comment by Strohgelaender** [comment]
@ge65cer I did not touch error.json in the other PR, they should not conflict each other. We also cannot add the prefix to error.json because it contains translations for non-Artemis error keys.

**Review by JohannesStoehr — DISMISSED** [review]
Checked the latest merge commit. Changes still look good

File: src/main/webapp/app/core/about-us/about-us.component.html [hunk 1]
```diff
@@ -18,18 +18,18 @@
                     <div class="col-md-4"><img class="img-fluid w-100 rounded" [src]="SERVER_API_URL + pm.photoDirectory" /></div>
                     <div class="col-md-8">
                         <h4 class="font-weight-bold">{{ pm.fullName }}</h4>
-                        <h5>{{ 'aboutUsOverview.' + pm.role | artemisTranslate }}</h5>
+                        <h5>{{ 'artemisApp.aboutUsOverview.' + pm.role | artemisTranslate }}</h5>
                         <span>
                             <i></i>
-                            <a href="{{ pm.website }}" target="_blank">{{ 'aboutUsOverview.website' | artemisTranslate }}</a>
+                            <a href="{{ pm.website }}" target="_blank">{{ 'artemisApp.aboutUsOverview.website' | artemisTranslate }}</a>
```
[the remark this answers is not in the document]
Comment (jpbernius) [reply src/main/webapp/app/core/about-us/about-us.component.html:24]: This is a valid comment, however, this is not related to the changes here and should be addressed in a follow-up.

**Comment by jpbernius** [comment]
Unfortunately, this PR broke two quiz-related test cases. Fixed in a4ed66d.

**Pull request merged**